ebos/ecloutputblackoilmodule.hh

```diff
@@ -1044,170 +1089,277 @@ public:
         }
     }
 
-    // write Fluid In Place to output log
-    void outputFipLog(std::map<std::string, double>& miscSummaryData,  std::map<std::string, std::vector<double>>& regionData, const bool substep)
-    {
-        const auto& comm = simulator_.gridView().comm();
-        auto maxElement = std::max_element(fipnum_.begin(), fipnum_.end());
-        size_t ntFip = 0;
-        if ( maxElement != fipnum_.end() ) {
-            ntFip = *maxElement;
+    int regionMax(const std::vector<int>& region) {
+        int max_value = 0;
+        const auto& comm = this->simulator_.gridView().comm();
+        auto maxElement = std::max_element(region.begin(), region.end());
+        if ( maxElement != region.end() ) {
```

Comment:
This can only fail if `region` is `empty()`.  I'd rather phrase this as
```C++
int regionMax(const std::vector<int>& region)
{
    const auto max_value = region.empty() ? 0
        : *std::max_element(region.begin(), region.end());

    return this->simulator_.gridView().comm().max(max_value);
}
```

Reply:
OK